packages/syft/src/syft/service/metadata/node_metadata.py

```diff
@@ -155,7 +155,7 @@ def check_version(self, client_version: str) -> bool:
         )
 
 
-@transform(NodeMetadata, NodeMetadataJSON)
+@transform(NodeMetadataV2, NodeMetadataJSON)
```

Comment:
@madhavajay had to rename this to NodeMetadataV2 and renamed NodeMetadataV1 to NodeMetadata because the current serde system cannot find a given class type from its canonical and version.

**Problem**
e.g.
If in a previous version data was stored under the class name NodeMetadata with version 1, the TYPE_BANK know how to serde this easily. But now in the new upgraded node NodeMetadata version 1 is renamed to NodeMetadataV1 and the class NodeMetadata now points to version 2. This becomes a problem during derserialization of old data from store. Since the data in store has object of NodeMetadata - version 1 but during deserde on upgraded node it tries to instantiate a NodeMetadata with version 2 since both of them have same FQN. 

Short Term Solution:
- Latest object version have different class names

Long term solution:

- For a given FQN Deserde identifies that object version is different than the current class present in TYPE_BANK and some kind of class type migration need to happen.
- Possibly a map of FQN to available versions